src/parser/SparqlParser.cpp

```diff
@@ -42,7 +42,12 @@ ParsedQuery SparqlParser::parse() {
 void SparqlParser::parseQuery(ParsedQuery* query, QueryType queryType) {
   if (queryType == CONSTRUCT_QUERY) {
     auto str = _lexer.getUnconsumedInput();
-    auto parseResult = sparqlParserHelpers::parseConstructTemplate(str);
+    SparqlQleverVisitor::PrefixMap prefixes;
+    for (const auto& prefix : query->_prefixes) {
+      prefixes[prefix._prefix] = prefix._uri;
+    }
+    auto parseResult =
+        sparqlParserHelpers::parseConstructTemplate(str, std::move(prefixes));
```

Comment:
Are the prefixes now parsed twice (for the CONSTRUCT clause and the WHERE clause)? I am asking because prefixes did work already for the WHERE clause.

Reply:
They are not, the weirdness comes from the parser-duality that currently exists within the codebase.
Basically the existing parser (the non-antlr) version currently parses all the prefixes, but the parsing of the construct clause is done using antlr (which requires the parsed prefixes being passed here), which then delegates back to the original parser to parse the where clause